examples/xgboost-quickstart/xgboost_quickstart/client_app.py

```diff
@@ -2,137 +2,110 @@
+    return bst
 
-def client_fn(context: Context):
+
+@app.train()
+def train(msg: Message, context: Context) -> Message:
     # Load model and data
     partition_id = context.node_config["partition-id"]
     num_partitions = context.node_config["num-partitions"]
-    train_dmatrix, valid_dmatrix, num_train, num_val = load_data(
-        partition_id, num_partitions
-    )
+    train_dmatrix, valid_dmatrix, num_train, _ = load_data(partition_id, num_partitions)
 
+    # Read from run config
+    num_local_round = context.run_config["local-epochs"]
+    # Flatted config dict and replace "-" with "_"
     cfg = replace_keys(unflatten_dict(context.run_config))
-    num_local_round = cfg["local_epochs"]
-
-    # Return Client instance
-    return FlowerClient(
-        train_dmatrix,
-        valid_dmatrix,
-        num_train,
-        num_val,
-        num_local_round,
-        cfg["params"],
-    )
+    params = cfg["params"]
+
+    global_round = msg.content["config"]["server-round"]
+    if global_round == 1:
+        # First round local training
+        bst = xgb.train(
+            params,
+            train_dmatrix,
+            num_boost_round=num_local_round,
+            evals=[(valid_dmatrix, "validate"), (train_dmatrix, "train")],
+        )
+    else:
+        bst = xgb.Booster(params=params)
+        global_model = bytearray(msg.content["arrays"]["0"].numpy().tobytes())
+
+        # Load global model into booster
+        bst.load_model(global_model)
+
+        # Local training
+        bst = _local_boost(bst, num_local_round, train_dmatrix)
+
+    # Save model
+    local_model = bst.save_raw("json")
+    model_np = np.frombuffer(local_model, dtype=np.uint8)
+
+    # Construct reply message
+    # Note: we store the model as the first item in a list into ArrayRecord,
+    # which can be accessed using index ["0"].
+    model_record = ArrayRecord([model_np])
+    metrics = {
+        "num-examples": num_train,
+    }
+    metric_record = MetricRecord(metrics)
+    content = RecordDict({"arrays": model_record, "metrics": metric_record})
+    return Message(content=content, reply_to=msg)
+
+
+@app.evaluate()
+def evaluate(msg: Message, context: Context) -> Message:
+    # Load model and data
+    partition_id = context.node_config["partition-id"]
+    num_partitions = context.node_config["num-partitions"]
+    _, valid_dmatrix, _, num_val = load_data(partition_id, num_partitions)
 
+    # Load config
+    cfg = replace_keys(unflatten_dict(context.run_config))
+    params = cfg["params"]
 
-# Flower ClientApp
-app = ClientApp(
-    client_fn,
-)
+    # Load global model
+    bst = xgb.Booster(params=params)
+    global_model = bytearray(msg.content["arrays"]["0"].numpy().tobytes())
+    bst.load_model(global_model)
+
+    # Run evaluation
+    eval_results = bst.eval_set(
+        evals=[(valid_dmatrix, "valid")],
+        iteration=bst.num_boosted_rounds() - 1,
+    )
+    auc = round(float(eval_results.split("\t")[1].split(":")[1]), 4)
```

Comment:
```suggestion
    auc = float(eval_results.split("\t")[1].split(":")[1])
```

No need to round, If there is a way to retrieve the value without parsing the string I think that's preferable. Maybe there is a callback ?

Reply:
`No need to round` --- I'll remove it.

`If there is a way to retrieve the value without parsing the string I think that's preferable` --- No, I didn't find a xgb built-in better way to do it